purchases/index.js

```diff
@@ -511,12 +511,18 @@ function listPurchases(db, options, callback) {
         callback = options;
         options = {};
     }
-    db.query(`SELECT * FROM purchase ORDER BY started_at DESC ${parseLimit(options)} ${parseOffset(options)}`, (error, result) => {
-        if(error) {
-            console.error("Failed to get purchases to list", error);
-            return callback(error);
+    db.query(`SELECT * FROM purchase ORDER BY started_at DESC ${parseLimit(options)} ${parseOffset(options)}`, (limitedError, limitedResult) => {
+        if(limitedError) {
+            console.error("Failed to get purchases to list", limitedError);
+            return callback(limitedError);
         }
-        callback(null, result.rows);
+        db.query(`SELECT * FROM purchase ORDER BY started_at DESC`, (error, result) => {
+            if(error) {
+                console.error("Failed to get purchases to list", error);
+                return callback(error);
+            }
+            callback(null, limitedResult.rows, result.rows.length);
+        })
```

Comment:
This is a terribly slow approach. You end up getting all the orders, serializing them and sending them to the server application, which then deserializes them, counts them, and responds with a number.

When possible always push the work to the database. Revert this commit, and instead look into SQL COUNT-expressions, c.f. e.g. https://stackoverflow.com/a/28916930/1080564